components/omega/doc/design/EOS.md

```diff
@@ -0,0 +1,181 @@
+Eos *Eos::create(const std::string &Name, const HorzMesh *Mesh, int NVertLevels, Config *Options);
+```
+
+#### 4.2.2 Initialization
+
+The init method will create the default eos and return an error code:
+```c++
+int Eos::init();
+```
+
+#### 4.2.3 Retrieval
+
+There will be methods for getting the default and non-default eos instances:
+```c++
+Eos *Eos::getDefault();
+Eos *Eos::get(const std::string &Name);
+```
+
+#### 4.2.4 Computation
+
+The public `computeSpecVol` method will rely on private methods for each specific EOS option (linear and TEOS-10).
+```c++
+void Eos::computeSpecVol(const Array2DReal &SpecVol, 
+                         const Array2DReal &ConservativeTemperature,
+                         const Array2DReal &AbsoluteSalinity,
+                         const Array2DReal &Pressure) {
+OMEGA_SCOPE(...)
+    if (eosChoice == EOSType::Linear){
+       parallelFor("eos-linear", {NCellsAll, NChunks},
+       KOKKOS_LAMBDA(int ICell, int KChunk) {
+          LocComputeSpecVolLinear(...);
+       });
+    } 
+    else if (eosChoice == EOSType::TEOS10Poly75t){
+       parallelFor("eos-teos10", {NCellsAll, NChunks},
+       KOKKOS_LAMBDA(int ICell, int KChunk) {
+          LocComputeSpecVolTEOS10Poly75t(...);
+        });
+    }...
+```
+
+
+
+#### 4.2.5 Destruction and removal
+No operations are needed in the destructor as the Kokkos arrays are removed
+when they are no longer in scope. The erase method
+will remove a named eos instance, whereas the clear method will remove all of
+them. Both will call the destructor in the process.
+```c++
+void Eos::erase(const std::string &Name);
+void Eos::clear();
+```
+## 5 Verification and Testing
+
+### 5.1 Test: Verification of GSW-C submodule
+A unit test will verify that the result of the GSW-C toolbox (used as a submodule, without modifications) matches the expected value of specific volume published in Roquet et al. 2015 within machine precision. The publication only includes one data point (Sa, Ct, P) to check. 
+
+### 5.2 Test: Verification of our TEOS-10 75-term polynomial implementation
+A unit test will verify that the result of the Omega implementation of the Roquet et al. 2015 75-term polynomial calculation of the specific volume matches the expected value of specific volume published in Roquet et al. 2015 within machine precision. The publication only includes one data point (Sa, Ct, P) to check. 
+
+### 5.3 Test: Verification of TEOS-10 with GSW-C toolbox
+A unit test will verify that the result of the Omega and GSW-C toolbox implementations for the Roquet et al. 2015 expansion are within machine precision over a range of conservative temperature, absolute salinity, and pressure ranges.
+
+### 5.4 Test: Verification of linear EOS
+The linear EOS should be close, but not equal (to machine precision)  to the TEOS-10 result. It should be exactly linear in T,S. It will be verified against known values from the MPAS-Ocean model for a range of temperature and salinity values.
```

Comment:
@sbrus89 and @alicebarthel, several questions come to mind here:
* When you say, "The linear EOS should be close, but not equal (to machine precision)  to the TEOS-10 result", do you mean for a reference T and S around which the EOS has been linearized?
* When you say "it should be exactly linear in T,S", is "it" the specific volume?
* If yes, a linear specific-volume EOS is not the same as a linear density EOS, so how will the comparison with MPAS-Ocean (which is linear in density) be performed?

Reply:
Thanks Xylar!

-  Yes, we typically do a check for the calculated value to match the published value (to 10^-10 a the moment) for the reference T, S input. A quick check for the linear EOS was that for the same input values, the calculated specific volume should be "close to" (e.g. 10^-2) the published value, but not of equal precision as TEOS-10 (10^-10). That was for a single value so maybe not that valuable but it was a mental check I did (i.e. a linear EOS sanity check) so I just added it in the unit tests.
- You are right, the "it" is density. (not specific volume)
- As mentioned above, the linear EOS implemented is an eos that is linear in density. It returns specific volume but the linearity (and check for linearity) is on the density. Which can be compared to MPAS-Ocean.